manager/src/main/java/org/openremote/manager/mqtt/TeltonikaMQTTHandler.java

```diff
@@ -0,0 +1,639 @@
+                String imeiString = imei.getValue().get();
+
+                // Get the device subscription information, and even if it's subscribed
+                TeltonikaDevice deviceInfo = connectionSubscriberInfoMap.get(imeiString);
+                //If it's null, the device is not subscribed, leave
+                if(deviceInfo == null) {
+                    //Maybe it does not have to be subscribed to the topic to receive the message?
+                    //When subscribed in MQTT Explorer, it works, but when not, deviceInfo is null
+
+                    getLogger().info(String.format("Device %s is not subscribed to topic, not posting message",
+                            imeiString));
+                    return;
+                // If it is subscribed, check that there is a command there, and send the command
+                } else{
+                    if(event.getValue().isPresent()){
+                        this.sendCommandToTeltonikaDevice((String)event.getValue().get(), deviceInfo);
+                    }
+                    else{
+                        getLogger().warning("Attribute sendToDevice was empty");
+                    }
+                }
+
+            }
+        }
+
+
+        getLogger().info("fired");
+    }
+
+    /**
+     * Sends a Command to the {@link TeltonikaDevice} in the correct format.
+     *
+     * @param command string of the command, without preformatting.
+     * List of valid commands can be found in Teltonika's website.
+     * @param device A {@link TeltonikaDevice} that is currently subscribed, to which to send the message to.
+     */
+    private void sendCommandToTeltonikaDevice(String command, TeltonikaDevice device) {
+        HashMap<String, String> cmdtest = new HashMap<>();
+        cmdtest.put("CMD", command);
+
+        mqttBrokerService.publishMessage(device.commandTopic, cmdtest, MqttQoS.EXACTLY_ONCE);
+    }
+
+    @Override
+    protected Logger getLogger() {
+        return LOG;
+    }
+    @Override
+    public boolean checkCanSubscribe(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        // Skip standard checks
+        if (!canSubscribe(connection, securityContext, topic)) {
+            getLogger().fine("Cannot subscribe to this topic, topic=" + topic + ", connection" + connection);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if the Subscribing client should be allowed to subscribe to the topic that is handled by this Handler.
+     * For Teltonika device endpoints, we need the fourth token (Index 3) to be a valid IMEI number.
+     * We do that by checking using IMEIValidator.
+     */
+    // To be removed when auto-provisioning works
+    @Override
+    public boolean canSubscribe(RemotingConnection connection, KeycloakSecurityContext securityContext, Topic topic) {
+        if(topic.getTokens().size() < 5){
```

Comment:
In other MQTT implementations I see the use of `topicMatches()` override.
Maybe worth practicing here too?